radix-engine-tests/tests/db/substate_database_overlay.rs

```diff
@@ -484,30 +484,59 @@ fn run_scenarios(
         .build();
 
     let mut next_nonce: u32 = 0;
-    for scenario_builder in get_builder_for_every_scenario() {
-        let epoch = ledger1.get_current_epoch();
-        let mut scenario = scenario_builder(ScenarioCore::new(network.clone(), epoch, next_nonce));
-        let mut previous = None;
-        loop {
-            let next = scenario
-                .next(previous.as_ref())
-                .map_err(|err| err.into_full(&scenario))
-                .unwrap();
-            match next {
-                NextAction::Transaction(next) => {
-                    let receipt1 = ledger1.execute_notarized_transaction(&next.raw_transaction);
-                    let receipt2 = ledger2.execute_notarized_transaction(&next.raw_transaction);
-
-                    check_callback(
-                        (ledger1.substate_db(), &receipt1),
-                        (ledger2.substate_db(), &receipt2),
-                    );
-
-                    previous = Some(receipt1);
-                }
-                NextAction::Completed(end_state) => {
-                    next_nonce = end_state.next_unused_nonce;
-                    break;
+    let mut scenario_builders = scenario_builders();
+    for requirement in ScenarioRequirements::all() {
```

Comment:
Is there any reason to apply protocol updates, then scenarios, then protocol updates, etc?

Could we instead just apply all protocol updates via the `LedgerSimulatorBuilder`, and then just run all scenarios?

Reply:
Mainly so that the scenarios are un against the protocol update that they were intended for to avoid any unexpected failures/successes that may happen if they're not run against their intended protocol update. 

With the refactored executor there is now ways to have finer grain control over these details, take a look at the tests in `dumper.rs`.